### PR TITLE
Add different skeletons for different strategies

### PR DESCRIPTION
**Pull request by mishogv**
Close: https://github.com/SoftUni-Internal/exam-systems-issues/issues/278
Close: https://github.com/SoftUni-Internal/exam-systems-issues/issues/281

### CODE REVIEW

File: Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/ProblemsController.cs [hunk 1]
```diff
@@ -979,5 +992,52 @@ private string GetOptimizedSolutionSkeleton(ProblemAdministrationViewModel probl
...
+                    });
+
+            problem.SubmissionTypes
+                .Where(
+                    x => !x.IsChecked &&
+                         x.Id.HasValue && 
+                         existingProblem.ProblemSubmissionTypesSkeletons
+                             .Any(y => y.SubmissionTypeId == x.Id.Value))
+                .ForEach(
+                x =>
```
Comment (shsimeonova) [inline Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/ProblemsController.cs:1035]: please align the code here

File: Open Judge System/Web/OJS.Web/Controllers/TempController.cs [hunk 2]
```diff
@@ -302,5 +304,42 @@ public ActionResult OptimizeMysqlProblemSkeletonAndTestInputQueries()
                 $"Updated {changedSkeletonsCount} solution skeletons. <br/>" +
                 $"Updated {changedTestsCount} test inputs.");
         }
+        
+        public ActionResult MigrateSolutionSkeletons()
+        {
+            var updatedProblemsCount = 0;
+            this.problemsDataService.GetAll()
+                .Where(x => x.SolutionSkeleton != null && x.SubmissionTypes.Any() && !x
+                .ProblemSubmissionTypesSkeletons.Any())
```
Comment (shsimeonova) [inline Open Judge System/Web/OJS.Web/Controllers/TempController.cs:313]: align code here so it is not on the same level as linq methods

File: Open Judge System/Web/OJS.Web/Controllers/TempController.cs [hunk 3]
```diff
@@ -302,5 +304,42 @@ public ActionResult OptimizeMysqlProblemSkeletonAndTestInputQueries()
...
+                                                SubmissionType = st,
+                                                SolutionSkeleton = p.SolutionSkeleton
+                                            }));
+
+                            this.problemsDataService.Update(p);
+                        }
+                        catch (Exception ex)
+                        {
+                            updatedProblemsCount--;
+                            Console.WriteLine(ex);
```
Comment (shsimeonova) [inline Open Judge System/Web/OJS.Web/Controllers/TempController.cs:338]: please use logger

File: Open Judge System/Web/OJS.Web/Scripts/Administration/Problems/problems-skeletons.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
+(() => {
```
Comment (shsimeonova) [inline Open Judge System/Web/OJS.Web/Scripts/Administration/Problems/problems-skeletons.js:1]: please check this for crlf endings

File: Open Judge System/Web/OJS.Web/Controllers/TempController.cs [hunk 5]
```diff
@@ -302,5 +304,42 @@ public ActionResult OptimizeMysqlProblemSkeletonAndTestInputQueries()
...
+            this.problemsDataService.GetAll()
+                .Where(x => x.SolutionSkeleton != null && x.SubmissionTypes.Any() && !x
+                .ProblemSubmissionTypesSkeletons.Any())
+                .ToList()
+                .ForEach(
+                    p =>
+                    {
+                        try
+                        {
+                            updatedProblemsCount++;
```
Comment (shsimeonova) [inline Open Judge System/Web/OJS.Web/Controllers/TempController.cs:320]: maybe move this after update so you don't have to decrease it in the catch block?

**Comment by shsimeonova** [comment]
Have we checked if the correct skeleton is sent to the distributor and remote workers?

**Comment by shsimeonova** [comment]
Please check all csthml and js files for correct crlf endings.

**Review by shsimeonova — CHANGES_REQUESTED** [review]

File: Open Judge System/Web/OJS.Web/Areas/Administration/Views/Problems/Edit.cshtml [hunk 6]
```diff
@@ -1,332 +1,333 @@
...
-                <strong>Case-insensitive</strong> @SharedResource.Case_insensitive_checker_description
-            </li>
-            <li>
-                <strong>Precision N</strong> @SharedResource.Precision_checker_description
-            </li>
-        </ul>
-    </script>
-    @Scripts.Render("~/bundles/jqueryval")
-}
+﻿@using System.Web.Mvc.Expressions;
```
Comment (Minkov) [inline Open Judge System/Web/OJS.Web/Areas/Administration/Views/Problems/Edit.cshtml:1]: A lot of changes here. Maybe line endings? Be sure to make them `crlf`

File: Open Judge System/Web/OJS.Web/Areas/Administration/Views/Problems/EditorTemplates/SubmissionTypeViewModel.cshtml [hunk 7]
```diff
@@ -1,22 +1,38 @@
...
-    <div class="editor-field col-xs-4">
-        @Html.EditorFor(m => m.IsChecked, new { @class = "form-control pull-left full-editor" })
-    </div>
-    <div class="editor-field col-xs-4">
-        <span class="glyphicon glyphicon-question-sign text-primary" title="@string.Format("{0} {1}", Resource.Allow_submission, Model.Name)" data-tooltip="true"></span>
-    </div>
-</div>
-
-
+﻿@using Resource = Resources.Areas.Administration.Problems.Views.EditorTemplates.SubmissionTypeViewModel;
```
Comment (Minkov) [inline Open Judge System/Web/OJS.Web/Areas/Administration/Views/Problems/EditorTemplates/SubmissionTypeViewModel.cshtml:1]: Make sure line endings are `crlf`

**Review by Minkov — CHANGES_REQUESTED** [review]

**Review by Minkov — APPROVED** [review]

**Review by shsimeonova — APPROVED** [review]

**Pull request merged**